apps/server/src/modules/deletion/controller/deletion-requests.controller.ts

```diff
@@ -0,0 +1,54 @@
+import { Body, Controller, Delete, Get, HttpCode, Param, Post, UseGuards } from '@nestjs/common';
+import { ApiOperation, ApiResponse, ApiTags } from '@nestjs/swagger';
+import { AuthGuard } from '@nestjs/passport';
+import { DeletionRequestUc } from '../uc';
+import { DeletionRequestLogResponse, DeletionRequestBodyProps, DeletionRequestResponse } from './dto';
+
+@ApiTags('DeletionRequests')
+@UseGuards(AuthGuard('api-key'))
+@Controller('deletionRequests')
+export class DeletionRequestsController {
+	constructor(private readonly deletionRequestUc: DeletionRequestUc) {}
+
+	@Post()
+	@HttpCode(202)
+	@ApiOperation({
+		summary: '"Queueing" a deletion request',
+	})
+	@ApiResponse({
+		status: 202,
+		type: DeletionRequestResponse,
+		description: 'Returns identifier of the deletion request and when deletion is planned at',
+	})
+	async createDeletionRequests(
+		@Body() deletionRequestBody: DeletionRequestBodyProps
+	): Promise<DeletionRequestResponse> {
+		return this.deletionRequestUc.createDeletionRequest(deletionRequestBody);
+	}
+
+	@Get(':requestId')
+	@HttpCode(200)
+	@ApiOperation({
+		summary: 'Retrieving details of performed or planned deletion',
+	})
+	@ApiResponse({
+		status: 200,
+		type: DeletionRequestLogResponse,
+		description: 'Return details of performed or planned deletion',
+	})
+	async getPerformedDeletionDetails(@Param('requestId') requestId: string): Promise<DeletionRequestLogResponse> {
```

Comment:
I do not get for what the create, performe and cancel is.

Reply:
I feel it needs to be discussed on a separate meeting, but speaking shortly:
- `createDeletionRequests()` method is for creating the deletion **request** so it will be queued and depending on the requested "delay" time, it should be then executed e.g. in 30 days. But this endpoint is only to create/queue the deletion request, not execute it directly in any way;
- `getPerformedDeletionDetails()` method returns the information about the queued deletion request - whether it has been already executed or not. If it was already executed, it will include the statistics about what changes has been made to the specific domain parts. If it was not yet executed, this endpoint will return an info on when it is expected to be executed;
- `cancelDeletionRequest()` method is for the possible cancelling the deletion request - what it does is, in fact, just deleting the previously queued deletion request (created by the `createDeletionRequests()` method call before).

CC @Metauriel

Reply:
ok